sos/cleaner/parsers/__init__.py

```diff
@@ -102,7 +103,9 @@ def parse_string_for_keys(self, string_data):
         :returns: The obfuscated line
         :rtype: ``str``
         """
-        for key, val in self.mapping.dataset.items():
+        for pair in sorted(self.mapping.dataset.items(), reverse=True,
+                           key=lambda x: len(x[0])):
+            key, val = pair
```

Comment:
This commit does not work in the scenario where `sos_commands/host/hostnamectl_status` has:

        Virtualization: kvm

When I run:

    python3 bin/sos report -o host --build --batch --clean --keywords Virt,Virtualization

I get:

        obfuscatedword2ualization: kvm

instead of:

        obfuscatedword3: kvm

Reply:
Hmm, that change is supposed to address that exact scenario. This will need some further digging.

Reply:
Oh! I see why. The keyword parser overrides `parse_line()` without eventually `super()`ing back into the base method.

That should be a quick fix, and is most likely the reason for the other problem below.

Reply:
This should be resolved in the most recent branch update.